### PR TITLE
Simplify by collapsing packages connection/ connectioncontext/ into networkservice/

### PR DESCRIPTION
**Pull request by edwarnicke**
Originally, connection and connectioncontext being separate from networkservice
were artifacts of the local/remote split in NSM.  This corrects that,
leaving us with two top level APIs:

1)  networkservice - Request/Close/MonitorConnection
2)  registry

### CODE REVIEW

**Comment by haiodo** [comment]
+1 from me

File: pkg/api/networkservice/mechanisms/vxlan/helpers.go [hunk 1]
```diff
@@ -21,8 +21,9 @@ import (
 
 	"github.com/pkg/errors"
 
-	"github.com/networkservicemesh/api/pkg/api/connection"
-	"github.com/networkservicemesh/api/pkg/api/connection/mechanisms/common"
+	"github.com/networkservicemesh/api/pkg/api/networkservice/mechanisms/common"
+
```
Comment (nickolaev) [inline pkg/api/networkservice/mechanisms/vxlan/helpers.go]: Nit: remove the blank line here.

File: pkg/api/networkservice/constants.go [hunk 2]
```diff
@@ -23,4 +23,9 @@ const (
 	PodNameKey = "podName"
 	// NamespaceKey - namespace a container is running in
 	NamespaceKey = "namespace"
+
+	// DNSConfigShouldNotBeNil -
+	DNSConfigShouldNotBeNil = "dnsConfig should not be nil"
+	// DNSServerIpsShouldHaveRecords -
+	DNSServerIpsShouldHaveRecords = "dnsConfig should have records"
```
Comment (nickolaev) [inline pkg/api/networkservice/constants.go]: Why we need these?

Comment (haiodo) [reply pkg/api/networkservice/constants.go]: @denis-tingajkin could you comment? @nickolaev  also I think we should not introduce fixes into this move pieces PR, but do all changes in separate ones. Thanks.

Comment (edwarnicke) [reply pkg/api/networkservice/constants.go]: Yeah... I have a lot of my own nits I want to fix too, but intentionally kept this a straight move

**Pull request merged**